src/py21cmfast/wrapper/inputs.py

```diff
@@ -1333,6 +1340,18 @@ def _astro_params_validator(self, att, val):
                     "update of M_TURN",
                     stacklevel=2,
                 )
+            if (
+                self.astro_options.USE_MASS_DEPENDENT_ZETA
+                and val.BETA_ESC != 0
+                and self.astro_options.PHOTON_CONS_TYPE
+                not in ["no-photoncons", "alpha-photoncons"]
+            ):
+                warnings.warn(
+                    f"You have set BETA_ESC != 0 but PHOTON_CONS_TYPE is {self.astro_options.PHOTON_CONS_TYPE}. "
+                    "This changes the escape fraction so it is not consistent with the manual setting of scaling."
+                    "Set PHOTON_CONS_TYPE to 'no-photoncons' or 'alpha-photoncons' if you want the scaling to be exact.",
+                    stacklevel=2,
+                )
```

Comment:
Someone else may have an opinion on this, but if there is a real problem using these two models together then perhaps we should throw an error instead of a warning

Reply:
I think an error will be better. Warnings are too easy to ignore.

Reply:
I don't understand is what is happening when we turn on `use_photoncons`, but I see in the code `f_esc10` gets modified per scaling relation if `PHOTON_CONS_TYPE == 3`. I will come back to it later when I understand the code better.